mongoose.c:2762:37: error: format specifies type 'unsigned short' but the argument has type 'int' [-Werror,-Wformat]
       inet_ntoa(sa->sin.sin_addr), ntohs(sa->sin.sin_port)));

We could either cast ntohs to an (unsigned short) or we can remove the -Werror
This patch does the former -- but there are other instances of ntohs not covered by this
kencu@macports.org




--- webserver/mongoose.c.orig	2019-02-25 20:37:28.000000000 -0800
+++ webserver/mongoose.c	2019-02-25 20:37:36.000000000 -0800
@@ -2759,7 +2759,7 @@
                                                 int proto,
                                                 union socket_address *sa) {
   DBG(("%p %s://%s:%hu", nc, proto == SOCK_DGRAM ? "udp" : "tcp",
-       inet_ntoa(sa->sin.sin_addr), ntohs(sa->sin.sin_port)));
+       inet_ntoa(sa->sin.sin_addr), (unsigned short)ntohs(sa->sin.sin_port)));
 
   nc->flags |= MG_F_CONNECTING;
   if (proto == SOCK_DGRAM) {
@@ -3473,7 +3473,7 @@
     int n =
         sendto(nc->sock, io->buf, io->len, 0, &nc->sa.sa, sizeof(nc->sa.sin));
     DBG(("%p %d %d %d %s:%hu", nc, nc->sock, n, mg_get_errno(),
-         inet_ntoa(nc->sa.sin.sin_addr), ntohs(nc->sa.sin.sin_port)));
+         inet_ntoa(nc->sa.sin.sin_addr), (unsigned short)ntohs(nc->sa.sin.sin_port)));
     mg_if_sent_cb(nc, n);
     return;
   }
